pandas/core/strings.py

```diff
@@ -1742,12 +1745,80 @@ def str_encode(arr, encoding, errors="strict"):
     return _na_map(f, arr)
 
 
-def _noarg_wrapper(f, docstring=None, **kargs):
+def forbid_nonstring_types(forbidden, name=None):
+    """
+    Decorator to forbid specific types for a method of StringMethods.
+
+    For calling `.str.{method}` on a Series or Index, it is necessary to first
```

Comment:
I thought about this a bit, and think that this is the cleanest solution - allow construction for the types that *may* be needed in a method, and then further restrict on a per-method level.

Reply:
so maybe a better way to do this (though slightly more breaking). is to have a ``.bytes`` accessor instead.